src/core/google-auth.js

```diff
@@ -26,44 +34,38 @@ XMPPSocialProvider.prototype.login = function(loginOpts, continuation) {
   }
 
   if (!this.credentials) {
-    this.oauth = freedom["core.oauth"]();
-    this.oauth.initiateOAuth(this.oAuthRedirectUris).then(function(stateObj) {
-      var oauthUrl = "https://accounts.google.com/o/oauth2/auth?" +
-               "client_id=" + this.oAuthClientId +
-               "&scope=" + this.oAuthScope +
-               "&redirect_uri=" + encodeURIComponent(stateObj.redirect) +
-               "&state=" + encodeURIComponent(stateObj.state) +
-               "&response_type=token";
-      var url = 'https://accounts.google.com/accountchooser?continue=' +
-          encodeURIComponent(oauthUrl);
-      return this.oauth.launchAuthFlow(url, stateObj);
-    }.bind(this)).then(function(continuation, responseUrl) {
-      var token = responseUrl.match(/access_token=([^&]+)/)[1];
-      var xhr = freedom["core.xhr"]();
-      xhr.open('GET', 'https://www.googleapis.com/oauth2/v1/userinfo?alt=json', true);
-      xhr.on("onload", function(continuation, token, xhr) {
-        xhr.getResponseText().then(function(continuation, token, responseText) {
-          var response = JSON.parse(responseText);
-          var credentials = {
-            userId: response.email,
-            jid: response.email,
-            oauth2_token: token,
-            oauth2_auth: 'http://www.google.com/talk/protocol/auth',
-            host: 'talk.google.com'
-          };
-          this.onCredentials(continuation, {cmd: 'auth', message: credentials});
-        }.bind(this, continuation, token));
-      }.bind(this, continuation, token, xhr));
-      xhr.setRequestHeader('Authorization', 'Bearer ' + token);
-      xhr.send();
-    }.bind(this, continuation)).catch(function (continuation, err) {
-      this.logger.error(err);
+    if (loginOpts.interactive && !loginOpts.rememberLogin) {
```

Comment:
No, I'm just keeping this so that clients of GTalk XMPP still have the option to just get access tokens without refresh tokens.

Also this is what will be used for chrome integration tests (https://github.com/freedomjs/freedom-social-xmpp/blob/master/spec/chrome_integration/gtalk.spec.js).  All of those tests expect that the oauth provider will return an access token.  There isn't a good way to change those to return a refresh token - we don't actually get the refresh token from oauth, rather we get a "code" from oauth which we then exchange for a refresh token , and we are only allowed to do this once per code.